test/e2e/update.go

```diff
@@ -41,3 +48,98 @@ var _ = Describe("Update clusters", func() {
 		Expect(oc.Tags).To(HaveKeyWithValue("key", &value))
 	})
 })
+
+var _ = Describe("Update cluster Managed Outbound IPs", func() {
+	var _ = BeforeEach(func(ctx context.Context) {
+		By("making sure clusters start with one IP")
+		oc, err := clients.OpenshiftClustersPreview.Get(ctx, vnetResourceGroup, clusterName)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("ensuring outbound-rule-4 has one IP")
+		lbName, err := getPublicLoadBalancerName(ctx)
+		Expect(err).NotTo(HaveOccurred())
+
+		rgName := stringutils.LastTokenByte(*oc.ClusterProfile.ResourceGroupID, '/')
+		lb, err := clients.LoadBalancers.Get(ctx, rgName, lbName, "")
+		Expect(err).NotTo(HaveOccurred())
+
+		if getOutboundIPsCount(lb) != 1 {
+			ocPatch := newManagedOutboundIPUpdateBody(1)
+			By("sending the PATCH request to set ManagedOutboundIPs.Count to 1")
+			err = clients.OpenshiftClustersPreview.UpdateAndWait(ctx, vnetResourceGroup, clusterName, ocPatch)
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
+
+	It("must be possible to increase and decrease IP Addresses on the public loadbalancer", func(ctx context.Context) {
+		By("sending the PATCH request to increase Managed Outbound IPs")
+		ocPatch := newManagedOutboundIPUpdateBody(5)
+		err := clients.OpenshiftClustersPreview.UpdateAndWait(ctx, vnetResourceGroup, clusterName, ocPatch)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("getting the cluster resource")
+		oc, err := clients.OpenshiftClustersPreview.Get(ctx, vnetResourceGroup, clusterName)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("checking effectiveOutboundIPs has been updated")
+		Expect(len(*oc.NetworkProfile.LoadBalancerProfile.EffectiveOutboundIps)).To(Equal(5))
+
+		By("checking outbound-rule-4 has required number IPs")
+		lbName, err := getPublicLoadBalancerName(ctx)
+		Expect(err).NotTo(HaveOccurred())
+
+		rgName := stringutils.LastTokenByte(*oc.ClusterProfile.ResourceGroupID, '/')
+		lb, err := clients.LoadBalancers.Get(ctx, rgName, lbName, "")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(getOutboundIPsCount(lb)).To(Equal(5))
```

Comment:
nit: We lookup lbName and rgName twice during test execution, even though this won't change over the duration of the test. This has a small potential to flake. Can we minimize that chance by only performing these lookups once, at the top of the test exec?